pkg/cri/sbserver/container_stats_list.go

```diff
@@ -115,6 +116,14 @@ func (c *criService) toCRIContainerStats(
 		if !ok {
 			handler, err = c.getMetricsHandler(ctx, cntr.SandboxID)
 			if err != nil {
+				// If the sandbox is not found, it may have been removed. we need to check container whether it is still exist
+				if errdefs.IsNotFound(err) {
+					_, err = c.containerStore.Get(cntr.ID)
```

Comment:
I know this is a backport, but this is masking the original error.
I'd expect we'd want the original error for line 127?

Reply:
should I fix this in main branch and then cherry-pick this into 1.7 as well?

Reply:
On second thought, the line on 127 should probably capture what line 121 errored out with if it was not  errdefs.IsNotFound(err) as it will probably give more insight into just the error not found

Reply:
I'd deal with it in main first.
This is likely fine as is, shouldn't break anything.